### PR TITLE
Update spring boot starter parent

### PR DESCRIPTION
**Pull request by IlyasBaqqari-CabinetOffice**
Updated spring boot starter parent to 3.1.5 to patch vulnerability CVE-2023-44487(7.5).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,6 +99,7 @@
 		<dependency>
 			<groupId>software.amazon.awssdk</groupId>
 			<artifactId>apigateway</artifactId>
+			<version>2.21.5</version>
```
Comment (GavCookCO) [inline pom.xml:102]: I think you can pass the `aws.java.sdk.version` variable into here instead of duplicating version numbers across dependencies.

Comment (IlyasBaqqari-CabinetOffice) [reply pom.xml:102]: Good shout, will change this.

Comment (Aloruss) [reply pom.xml]: in this case you don't need to specify the version, as in dependencyManagement(line 23) will take care of all aws dependency

Comment (a-lor-cab) [reply pom.xml]: in this case you don't need to specify the version, as in dependencyManagement(line 23) the amazone bom will take care of all aws dependency

Comment (IlyasBaqqari-CabinetOffice) [reply pom.xml]: Nice, I'll just remove that line 👍

**Review by iaincooper-tco — APPROVED** [review]

**Pull request merged**